### ISSUE TITLE
Generic tuple parameter can't be concretised in loop

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1189]
**Describe the bug**
When a tuple is parametrised with a type variable and the replacement type is known, it can't be concretised in a loop; that is, the type of the loop variable elements of the tuple are assigned to is not a concrete type.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That `c` would be `str` in both instances; that `s` in `qux` would be `str`.

**Screenshots or Code**
```python
# pyright: strict

foo = ["foo"]

def bar() -> None:
    for s in zip(foo):
        reveal_type(s)  # Type of "s" is "Tuple[str]"
        for c in s:
            reveal_type(c)  # Type of "c" is "_T1"
        (reveal_type(c) for c in s)  # Type of "c" is "_T1"


from typing import Iterable, Tuple, TypeVar

T = TypeVar("T")

def baz(v: Iterable[T]) -> Tuple[T]: ...

def qux() -> None:
    quux = baz(foo)
    reveal_type(quux)  # Type of "quux" is "Tuple[str]"
    reveal_type(quux[0])  # Type of "quux[0]" is "str"
    for s in quux:
        reveal_type(s)  # Type of "s" is "T"
        for c in s:
            reveal_type(c)  # Type of "c" is "Any"

```

**VS Code extension or command-line**
Either, 1.1.87

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.88, which I just published. It will also be included in the next release of Pylance.
